code/game/objects/items/storage/belt.dm

```diff
@@ -685,7 +685,10 @@
 		sure that we don't have to do any extra calculations.
 		*/
 		playsound(src,drawSound, 15, 1)
-		gun_underlay = image(icon, src, current_gun.icon_state)
+		if(current_gun.greyscale_config && current_gun.greyscale_colors)
+			gun_underlay = icon(SSgreyscale.GetColoredIconByType(current_gun.greyscale_config, current_gun.greyscale_colors), current_gun.rounds ? "belt" : "belt_empty")
```

Comment:
Why do you use icon() here instead of image or MA its a lot more expensive for no reason